src/firebolt/async_db/cursor.py

```diff
@@ -469,12 +469,15 @@ async def _api_request(
                 self._set_parameters to be ignored.
         """
         parameters = parameters or {}
+        account_version = await self._client._account_version
         if use_set_parameters:
             parameters = {**(self._set_parameters or {}), **parameters}
         if self.parameters:
             parameters = {**self.parameters, **parameters}
-        # Engines v2 always require account_id
-        if self.connection._is_system or (await self._client._account_version) == 2:
+        # Engines v2 system engine will supply account_id dynamically in the url
+        if (self.connection._is_system and account_version == 1) or (
```

Comment:
Why do we have a `not self.connection._is_system and account_version == 2` branch? Does this mean we would add `account_id` to all user engines for v2 accounts?

Reply:
I believe this condition should just be `if self.connection._is_system and account_version == 1`

Reply:
Do we not need to add `account_id` to user engines anymore? Reading https://packboard.atlassian.net/browse/FIR-31836 it specifies only system engine. If that's the case, I'm more than happy to remove this clause and simplify the expression.